unit_test.py

```diff
@@ -0,0 +1,1371 @@
+        self.assertEqual(experiment_query.value_type, value_type)
+        self.assertEqual(experiment_query.value_range, value_range)
+        self.assertEqual(experiment_query.value_candidates,
+                         value_candidates)
+        return experiment_query
+
+    def test_create_observation(self, experiment_id=rand_experiment_id()):
+        student_name = randlength_word()
+        datum = randlength_word()
+        observation_id = u.generate_observation_id(experiment_id, student_name)
+        while ds.query(m.Observation).filter(m.Observation.id == observation_id).count() != 0:
+            student_name = randlength_word()
+            experiment_id = rand_experiment_id()
+            observation_id = u.generate_observation_id(experiment_id,
+                                                       student_name)
+        observation = m.Observation(id=observation_id,
+                                    student_name=student_name,
+                                    datum=datum)
+
+        ds.add(observation)
+        ds.commit()
+        observation_query = ds.query(m.Observation).filter(m.Observation.id == observation_id).first()
+        self.assertEqual(observation_query.id, observation_id)
+        self.assertEqual(observation_query.student_name, student_name)
+        self.assertEqual(observation_query.datum, datum)
+        return observation_query
+
+    def test_create_user(self):
+        username = randlength_word()
+        while ds.query(m.User).filter(m.User.id == username).count() != 0:
+            username = randlength_word()
+        name = randlength_word()
+        user = m.User(id=username,
+                      name=name)
+        ds.add(user)
+        ds.commit()
+        user_query = ds.query(m.User).filter(m.User.id == username).first()
+        self.assertEqual(user_query.id, username)
+        self.assertEqual(user_query.name, name)
+        return user_query
+
+    def test_create_role(self):
+        name = randlength_word()
+        while ds.query(m.Role).filter(m.Role.name == name).count() != 0:
+            name = randlength_word()
+        powers = rand_powers()
+        role = m.Role(name=name, powers=powers)
+        ds.add(role)
+        ds.commit()
+        role_query = u.get_role(name)
+        self.assertEqual(role_query.name, name)
+        self.assertEqual(role_query.powers, powers)
+        return role_query
+
+    def test_create_power(self):
+        id = randlength_word()
+        while ds.query(m.Power).filter(m.Power.id == id).count() != 0:
+            id = randlength_word()
+        power = m.Power(id=id)
+        ds.add(power)
+        ds.commit()
+        power_query = u.get_power(id)
+        self.assertEqual(power_query.id, id)
+        return power_query
+
+    # --- Testing functions(have interaction with database) for functions in utility.py ---
+
+    # Testing for the existence of an object
+    # This group of functions create a new object in the database with random
```

Comment:
this kind of comment is unnecessary. You're comments should only describe weird or surprising things about your code, not what the code itself is doing.